### ISSUE TITLE
No good way to get PresamplePackages in correct order from Campaign

### ISSUE THREAD

**Issue by PascalLesage** [PascalLesage/brightway2-presamples#43]
Multiple presample packages can be passed to e.g. MonteCarloLCA
If an element (a parameter, a matrix index) is part of more than one package, the value in the package that is later in the list is the one that is used.
Knowing this, there should be a way to generate a list of presample resources from campaign that is ordered thus: 
- Ordered resources from oldest parent
- Ordered resources from second oldest parent
...
- Ordered resources from direct parent
- Ordered resources from self.

Currently, Campaign.ancestors will list ancestors in the reverse of the desired order (from direct parent to furthest ancestor). It also does not list presample resources, just campaigns.  

The solution should be a class method that looks something like this:
```python
def get_all_resources_in_proper_order(self):    
    ancestors = list(self.ancestors)
    ancestors.reverse()
    resources = [p.path for ancestor in ancestors for p in ancestor.packages]
    try:
        resources.append(*[p.path for p in self.packages])
    except:
        pass
    return resources
```
@cmutel I tag you in case this is already implemented somewhere and I just can't find it.

**Comment by cmutel** [comment]
Just to clarify - you would want the list of resources, but are not asking for us to determine which resources could (potentially) overwrite previous ones, right?

Also, I think you want [extend instead of append](https://stackoverflow.com/questions/252703/difference-between-append-vs-extend-list-methods-in-python) :)

**Comment by PascalLesage** [comment]
> Just to clarify - you would want the list of resources, but are not asking for us to determine which resources could (potentially) overwrite previous ones, right?
Indeed. The idea is to pass the list of presamples filepaths in the correct order to e.g. MonteCarloLCA instantiation.
